agent/consul/multilimiter/multilimiter_test.go

```diff
@@ -30,58 +31,168 @@ func TestNewMultiLimiter(t *testing.T) {
 }
 
 func TestRateLimiterUpdate(t *testing.T) {
-	c := Config{LimiterConfig: LimiterConfig{Rate: 0.1}, ReconcileCheckLimit: 1 * time.Millisecond, ReconcileCheckInterval: 10 * time.Millisecond}
+	c := Config{LimiterConfig: LimiterConfig{Rate: 0.1}, ReconcileCheckLimit: 1 * time.Hour, ReconcileCheckInterval: 10 * time.Millisecond}
 	m := NewMultiLimiter(c)
+	m.Run(context.Background())
 	key := makeKey([]byte("test"))
+
+	//Allow a key
 	m.Allow(Limited{key: key})
+	time.Sleep(100 * time.Millisecond)
 	limiters := m.limiters.Load()
 	l1, ok1 := limiters.Get(key)
+
+	// check key exist
 	require.True(t, ok1)
 	require.NotNil(t, l1)
+
 	la1 := l1.(*Limiter).lastAccess.Load()
+
+	// allow same key again
+	time.Sleep(10 * time.Millisecond)
 	m.Allow(Limited{key: key})
 	limiters = m.limiters.Load()
 	l2, ok2 := limiters.Get(key)
+
+	// check it exist and it's same key
 	require.True(t, ok2)
 	require.NotNil(t, l2)
 	require.Equal(t, l1, l2)
+
+	// last access should be different
 	la2 := l1.(*Limiter).lastAccess.Load()
-	require.Equal(t, la1, la2)
+	require.NotEqual(t, la1, la2)
 
 }
 
 func TestRateLimiterCleanup(t *testing.T) {
 
 	// Create a limiter and Allow a key, check the key exists
-	c := Config{LimiterConfig: LimiterConfig{Rate: 0.1}, ReconcileCheckLimit: 1 * time.Millisecond, ReconcileCheckInterval: 10 * time.Millisecond}
+	c := Config{LimiterConfig: LimiterConfig{Rate: 0.1}, ReconcileCheckLimit: 1 * time.Second, ReconcileCheckInterval: 10 * time.Millisecond}
 	m := NewMultiLimiter(c)
+	limiters := m.limiters.Load()
 	ctx, cancel := context.WithCancel(context.Background())
 	defer cancel()
 	m.Run(ctx)
 	key := makeKey([]byte("test"))
 	m.Allow(Limited{key: key})
-	limiters := m.limiters.Load()
+	retry.Run(t, func(r *retry.R) {
+		time.Sleep(100 * time.Millisecond)
+		l := m.limiters.Load()
+		require.NotEqual(r, limiters, l)
+		limiters = l
+	})
```

Comment:
The `retry` package has a method that lets you configure the interval between each check:

```suggestion
	retry.RunWith(&retry.Timer{Wait: 100 * time.Millisecond, Timeout: 2 * time.Second}, t, func(r *retry.R) {
		time.Sleep(100 * time.Millisecond)
		l := m.limiters.Load()
		require.NotEqual(r, limiters, l)
		limiters = l
	})
```